File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            List<Data> list = new List<Data>();
            Random random = new Random();

            for (int i = 0; i < 10; i++)
            {
                Data data = new Data(string.Format("Name {0}", i), string.Format("Category {0}", random.Next(4)), random.Next(1000));
                list.Add(data);
            }

            XtraReport1 report = new XtraReport1(list, Convert.ToInt32(spinEdit1.EditValue));
            report.ShowPreviewDialog();
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

Namespace WindowsFormsApplication1

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim list As List(Of Data) = New List(Of Data)()
            Dim random As Random = New Random()
            For i As Integer = 0 To 10 - 1
                Dim data As Data = New Data(String.Format("Name {0}", i), String.Format("Category {0}", random.Next(4)), random.Next(1000))
                list.Add(data)
            Next

            Dim report As XtraReport1 = New XtraReport1()
            report.DataSource = list
            report.ShowPreviewDialog()
        End Sub
    End Class
End Namespace
